src/fcntl.rs

```diff
@@ -380,3 +380,20 @@ pub fn fallocate(fd: RawFd, mode: FallocateFlags, offset: libc::off_t, len: libc
     let res = unsafe { libc::fallocate(fd, mode.bits(), offset, len) };
     Errno::result(res)
 }
+
+/// Allocate file space.
+///
+/// Ensure disk space is allocated for the file referred to by file
+/// descriptor `fd` for the bytes in the range starting at `offset` and
+/// continuing for `len` bytes.
+///
+/// # References
+///
+/// [`posix_fallocate`](http://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_fallocate.html)
+#[cfg(any(target_os = "android", target_os = "dragonfly", target_os = "freebsd",
```

Comment:
`posix_fallocate` is not defined in `libc` for DragonFlyBSD and OpenBSD and I couldn't find it in the public sources for those OSes either. I think this and the tests should be removed for those platforms.